cms/private_media/views.py

```diff
@@ -0,0 +1,34 @@
+from typing import TYPE_CHECKING
+
+from django.core.exceptions import PermissionDenied
+from django.shortcuts import redirect
+from wagtail.images.permissions import permission_policy
+from wagtail.images.views.serve import ServeView
+
+if TYPE_CHECKING:
+    from django.http import HttpResponse
+
+    from cms.private_media.models import AbstractPrivateRendition, PrivateImageMixin
+
+
+class PrivateImageServeView(ServeView):
+    def serve(self, rendition: "AbstractPrivateRendition") -> "HttpResponse":
+        image: PrivateImageMixin = rendition.image
+
+        # If there's no reason (within our control) for the file not to be served by
+        # media infrastructure, redirect
+        if image.is_public and not image.file_permissions_are_outdated():
+            return redirect(rendition.file.url)
+
+        # Block access to private images if the user has insufficient permissions
+        user = self.request.user
+        if image.is_private and (
+            not user.is_authenticated
+            or not permission_policy.user_has_any_permission_for_instance(user, ["choose", "add", "change"], image)
+        ):
+            raise PermissionDenied
+
+        # Serve the file until it is no longer private, or file permissions
+        # have been set successfully
+        response: HttpResponse = super().serve(rendition)
```

Comment:
thought: could we serve this without any cache. It is for authenticated users with permissions after all. This way we'd only need to purge `rendition.file.url`s should anything change for the public files.

Reply:
ah, but the manager `bulk_set_privacy` invalidates FE cache. Still, we could eliminate `serve_url_batch.purge()` from that 🤔

Reply:
> thought: could we serve this without any cache. It is for authenticated users with permissions after all. This way we'd only need to purge rendition.file.urls should anything change for the public files

I'm open to this. This approach is certainly a good idea whilst the media is still private. Once public, caching is probably more desirable, but 99% of the time, the image serve view should no longer receive hits, because the `src` value on images should use the direct file URL.

Reply:
Implemented in [19dde02](https://github.com/ONSdigital/dis-wagtail/pull/46/commits/19dde02607d9cae65100c8ceac05407b035ffc65)